cmd/apiserver-watcher/run.go

```diff
@@ -122,6 +134,9 @@ func runRunCmd(cmd *cobra.Command, args []string) error {
 			if err := handler.onFailure(); err != nil {
 				glog.Infof("Failed to mark service down on signal: %s", err)
 			}
+			if err := handler.lockfile.Unlock(); err != nil {
```

Comment:
I couldn't put much though on this sorry, but I have the feeling this can go wrong in several places, just brainstorming here:

1. we get the lock on start and fail if we can't
2. we ONLY unlock on signal received

I don't know if there are some edge case we can exit without unlocking so we never will be able to start.

In case we always unlock on exit, a parallel pod restart will release the lock, despite the original pod is still active, since it never checks again if he has the lock ...

Reply:
Since we're in HostPID namespace, we'll do the right thing and remove stale locks if the holding process is dead. I think we should be safe in this regard.

Reply:
ahh, smart
> // Please note, that existing lockfiles containing pids of dead processes
> // and lockfiles containing no pid at all are simply deleted.